lib/logger.js

```diff
@@ -0,0 +1,94 @@
+const { HyperCoreLogger } = require("hypercore-logs")
+
+class Logger {
+  constructor(server, opts = {}) {
+    if(!server || !(server instanceof HyperCoreLogger)) {
+      throw new Error('server is required in logger constructor')
+    }
+
+    this.server = server
+
+    const defaultOpts = {
+      enableConsole: true,
+      enableHypercore: true
+    }
+
+    this.opts = {
+      ...defaultOpts,
+      ...opts
+    }
+  }
+
+  log (...params) {
+    const line = this._formatMessage(params)
+    this._logHypercore(line)
+
+    if (this.opts.enableConsole) {
+      console.log(line)
+    }
+  }
+
+  info (...params) {
+    return this.log(...params)
+  }
+
+  error (...params) {
+    const line = this._formatMessage(params, true)
+    this._logHypercore(line)
+
+    if (this.opts.enableConsole) {
+      console.error(line)
+    }
+  }
+
+  trace (...params) {
+    const line = this._formatMessage(params, true)
+    const msg = `${line}\n${(new Error()).stack}`
+
+    this._logHypercore(msg)
+
+    if (this.opts.enableConsole) {
+      console.error(msg)
```

Comment:
we could use console.trace here

Reply:
The stack trace will be included as part of the `msg` on line 46.
Is there any other reason you suggested using `console.trace` here?

I opted for not using `trace` to make sure the hypercore and console messages are consistent, including the stack trace. Otherwise they would be different.

Reply:
btw, console.log() uses within console.trace() for Error objects.